changelog/fragments/k8s-1.24-bump.yaml

```diff
@@ -0,0 +1,110 @@
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Go (go/v3) and Hybrid Helm (hybrid) Operators - Upgrading to Kubernetes 1.24
+      body: |
+        To update your project to support Kubernetes 1.24:
+        
+        1. In `go.mod`:
+
+        - Replace `k8s.io/api v0.23.5` with `k8s.io/api v0.24.0`
+        - Replace `k8s.io/apimachinery v0.23.5` with  `k8s.io/apimachinery v0.24.0`
+        - Replace `k8s.io/client-go v0.23.5` with `k8s.io/client-go v0.24.0`
+        - Replace `sigs.k8s.io/controller-runtime v0.11.2` with `sigs.k8s.io/controller-runtime v0.12.1`
+        - Hybrid Helm (hybrid) only: Replace `github.com/operator-framework/helm-operator-plugins v0.0.11` with `github.com/operator-framework/helm-operator-plugins v0.0.12-0.20220608155702-d3967d2ae2ac`
+
+        2. Run `go mod tidy` to ensure all dependencies are upgraded
+        
+        3. In `Makefile`:
+        - Replace `CONTROLLER_TOOLS_VERSION ?= 0.8.0` with `CONTROLLER_TOOLS_VERSION ?= 0.9.0`
+        - Replace `ENVTEST_K8S_VERSION = 1.23` with `ENVTEST_K8S_VERSION = 1.24`
+
+        4. Run `make` to run all make targets
+
+  - description: >
+      For Go (go/v3) and Hybrid Helm (hybrid) Operators: Support for Go 1.18
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Go (go/v3) and Hybrid Helm (hybrid) Operator - Update to Go 1.18
+      body: |
+        To update your project to use Go 1.18:
+
+        1. In the `go.mod` file:
+
+        - Replace `go 1.17` with `go 1.18`   
+
+        2. Run `go mod tidy` to ensure all dependencies are upgraded
+
+  - description: >
+      For `operator-sdk bundle validate`: fix channel naming validation [More info](https://github.com/operator-framework/api/pull/231)
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "bugfix"
+
+    # Is this a breaking change?
+    breaking: false
+
+  - description: >
+      For `operator-sdk bundle validate`: Add CRD description check validation [More info](https://github.com/operator-framework/api/pull/234)
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "addition"
+
+    # Is this a breaking change?
+    breaking: false
+
+  - description: >
+      For `operator-sdk bundle validate`: Add a new check to warn authors when permissions to create CRDs are found [More info](https://github.com/operator-framework/api/pull/241)
+      
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
```

Comment:
Also, I forget to put in the latest PR to bump KB and we need to add here;

```
  - description: >
      For Java (ava/v1alpha) based-language operator: Upgrade dependence from `v0.0.3` to `v0.5.1`. [More info](https://github.com/operator-framework/java-operator-plugins/releases)
    breaking: false    
```

Reply:
But that works without old ones as well usually. ideally, we need to have them all aligned to avoid some unexpected scenarios.

Reply:
commit https://github.com/operator-framework/operator-sdk/pull/5843/commits/1ccecca4e08393c453999d188a1bc92e3a3fa549